### PR TITLE
feat: pattern policy-insights\remediation - `avm/ptn/policy-insights/remediation`

### PR DESCRIPTION
**Pull request by donk-msft**
## Description

Migrated policy-insights\remediation from CARML to new AVM pattern.

## Pipeline Reference

| Pipeline |
| -------- |
|   [avm.ptn.policy-insights.remediation](https://github.com/donk-msft/bicep-registry-modules/actions/workflows/avm.ptn.policy-insights.remediation.yml)    |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [X] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by donk-msft** [comment]
Pipeline overall successful, but failed on removal of remediation task. Seems timing issue.

image

File: avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,152 @@
+targetScope = 'subscription'
+
+metadata name = 'Using only defaults'
```
Comment (eriqua) [inline avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep:3]: Please update metadata name and description to waf-aligned related tests (you can reference any res module)

File: avm/ptn/policy-insights/remediation/tests/e2e/max/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,152 @@
+targetScope = 'subscription'
+
+metadata name = 'Using only defaults'
```
Comment (eriqua) [inline avm/ptn/policy-insights/remediation/tests/e2e/max/main.test.bicep:3]: Please update metadata name and description to max test related values (you can reference any res module)

File: avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep (hunk 1 above)
Comment (eriqua) [reply avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep]: The description is still to be updated. Also please do not resolve comments yourself, it makes the review more difficult 😉

File: avm/ptn/policy-insights/remediation/tests/e2e/max/main.test.bicep (hunk 2 above)
Comment (eriqua) [reply avm/ptn/policy-insights/remediation/tests/e2e/max/main.test.bicep]: Same here, please update the description as well

File: avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep (hunk 1 above)
Comment (donk-msft) [reply avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep]: Changed to:
metadata name = 'WAF-aligned'
metadata description = 'This instance deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.'

**Review by ChrisSidebotham — DISMISSED** [review]
Looks good @donk-msft - Please can you attach the pipeline badge for the workflow as discussed when we spoke in person :)

File: avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep (hunk 1 above)
Comment (eriqua) [reply avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep]: Thank you! Please also rerun the Set-AVMModule utility for changes to be reflected in the Readme as well

File: .github/workflows/avm.ptn.policy-insights.remediation.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        default: true
+  push:
+    branches:
+      - main
+    paths:
+      - ".github/actions/templates/avm-**"
+      - ".github/workflows/avm.template.module.yml"
+      - ".github/workflows/avm.ptn.policy-insights.remediation.yml"
+      - "avm.ptn.policy-insights.remediation/**"
+      - "avm/utilities/pipelines/**"
```
Comment (eriqua) [inline .github/workflows/avm.ptn.policy-insights.remediation.yml:31]:
```suggestion
      - "avm/utilities/pipelines/**"
      - "!avm/utilities/pipelines/platform/**"
```

Comment (donk-msft) [reply .github/workflows/avm.ptn.policy-insights.remediation.yml:31]: Added suggested changes

File: avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep (hunk 1 above)
Comment (donk-msft) [reply avm/ptn/policy-insights/remediation/tests/e2e/waf-aligned/main.test.bicep]: Updated readme

**Comment by donk-msft** [comment]
@eriqua is anything still blocking approval?

File: avm/ptn/policy-insights/remediation/management-group/main.bicep
Comment (eriqua) [inline avm/ptn/policy-insights/remediation/management-group/main.bicep:1]: We opted for a cleaner structure for ptn modules wrt the one from CARML. Please refer to https://github.com/Azure/bicep-registry-modules/tree/main/avm/ptn/authorization/policy-assignment

- create a modules folder
- move management-group/main.bicep there and rename to management-group.bicep
- move subscription/main.bicep there and rename to subscription.bicep
- move resource-group/main.bicep there and rename to resource-group.bicep
- Remove corresponding main.json and readme files
- Update test files accordingly.

Comment (donk-msft) [reply avm/ptn/policy-insights/remediation/management-group/main.bicep:1]: @eriqua comments and issue with set-avmmodule are resolved.

File: avm/ptn/policy-insights/remediation/tests/e2e/mg.defaults/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+// ============== //
+
+module testDeployment '../../../main.bicep' = {
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: resourceLocation
+    policyAssignmentId: policySetAssignments.id
+    policyDefinitionReferenceId: 'Prerequisite_DeployExtensionWindows'
+    filtersLocations: []
```
Comment (eriqua) [inline avm/ptn/policy-insights/remediation/tests/e2e/mg.defaults/main.test.bicep:48]: Defaults tests should only include input parameters of type required. They should leverage default values for all the others. Please clean up optional parameters from the 3 default tests where applicable. E.g., in this case:
```suggestion
```

Comment (donk-msft) [reply avm/ptn/policy-insights/remediation/tests/e2e/mg.defaults/main.test.bicep:48]: Updated

Comment (eriqua) [reply avm/ptn/policy-insights/remediation/tests/e2e/mg.defaults/main.test.bicep]: Hey @donk-msft, that's actually the case for all defaults tests if you can.
Everything else looks great. Thanks a lot!

Comment (donk-msft) [reply avm/ptn/policy-insights/remediation/tests/e2e/mg.defaults/main.test.bicep]: I updated all default tests... not sure what you mean.

File: avm/ptn/policy-insights/remediation/tests/e2e/sub.defaults/main.test.bicep [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+module testDeployment '../../../main.bicep' = {
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: resourceLocation
+    subscriptionId: subscriptionId
+    policyAssignmentId: policySetAssignments.outputs.resourceId
+    policyDefinitionReferenceId: 'Prerequisite_DeployExtensionWindows'
+    filtersLocations: []
```
Comment (eriqua) [inline avm/ptn/policy-insights/remediation/tests/e2e/sub.defaults/main.test.bicep:51]: For example here I also see not required parameters used

File: avm/ptn/policy-insights/remediation/tests/e2e/mg.defaults/main.test.bicep (hunk 4 above)
Comment (eriqua) [reply avm/ptn/policy-insights/remediation/tests/e2e/mg.defaults/main.test.bicep]: I just pointed it out directly in the other test file. The subscription defaults seems to still use the empty filtersLocations array

File: avm/ptn/policy-insights/remediation/tests/e2e/sub.defaults/main.test.bicep (hunk 5 above)
Comment (donk-msft) [reply avm/ptn/policy-insights/remediation/tests/e2e/sub.defaults/main.test.bicep]: @eriqua - apologies, how could I've overlooked that. Made the changes.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**